### PR TITLE
Update search docs: OpenSearch support and backlog 790

### LINKED ISSUE #790: Describe OpenSearch support
See https://github.com/jmix-framework/jmix/issues/2094

### PR DESCRIPTION
**Pull request by Gavrilov-Ivan**
See #790
Also includes the following issues with content actual for 2.3:

- #245 
- #246 
- #247 
- #251

### CODE REVIEW

File: content/modules/search/pages/config-options.adoc [hunk 1]
```diff
@@ -77,4 +79,33 @@ jmix.search.changedEntitiesIndexingEnabled = false
 To utilize the functionality of the Search add-on, users with limited access to the system should have one of the following roles:
 
 * *Search: edit filter* provides permissions to add full-text search conditions to the filter.
-* *Search: view search results* provides permissions to access the Search Results view.
\ No newline at end of file
+* *Search: view search results* provides permissions to access the Search Results view.
+
+[[index-settings]]
+== Index Settings and Analysis
+Add-on allows to configure index settings including analysis.
```
Comment (knstvk) [inline content/modules/search/pages/config-options.adoc]: "The add-on allows you to ..."

File: content/modules/search/pages/config-options.adoc [hunk 2]
```diff
@@ -77,4 +79,33 @@ jmix.search.changedEntitiesIndexingEnabled = false
...
+
+NOTE: This configuration depends on used search platform (OpenSearch or Elasticsearch) and should be recreated accordingly if you switch between them.
+
+* Create spring bean that implements `OpenSearchIndexSettingsConfigurer` (or `ElasticsearchIndexSettingsConfigurer` in case of Elasticsearch usage).
```
Comment (knstvk) [inline content/modules/search/pages/config-options.adoc]: "Create a Spring bean that ..."

File: content/modules/search/pages/config-options.adoc [hunk 3]
```diff
@@ -77,4 +79,33 @@ jmix.search.changedEntitiesIndexingEnabled = false
...
+* Implement method `configure(OpenSearchIndexSettingsConfigurationContext)`.
+
+=== Implementation
```
Comment (knstvk) [inline content/modules/search/pages/config-options.adoc:94]: Add an anchor

File: content/modules/search/pages/config-options.adoc [hunk 4]
```diff
@@ -77,4 +79,33 @@ jmix.search.changedEntitiesIndexingEnabled = false
...
+
+Settings can be configured via settings builder (belongs to OpenSearch\Elasticsearch Java API client).
```
Comment (knstvk) [inline content/modules/search/pages/config-options.adoc]: Use forward slash here and in other places: OpenSearch/Elasticsearch

File: content/modules/search/pages/config-options.adoc [hunk 5]
```diff
@@ -77,4 +79,33 @@ jmix.search.changedEntitiesIndexingEnabled = false
...
+
+Configuration context provides two types of settings builder:
+
+* *Common* - allows to configure settings that will be applied to all indexes. Can be acquired via `context.getCommonSettingsBuilder()`.
```
Comment (knstvk) [inline content/modules/search/pages/config-options.adoc]: Always "allows [someone] to" - fix everywhere

File: content/modules/search/pages/index-definitions.adoc [hunk 6]
```diff
@@ -9,13 +9,18 @@ A Java interface defining an index must adhere to these requirements:
...
 * It can be named arbitrarily.
 * It must be annotated with the `@JmixEntitySearchIndex` annotation.
-* The annotation should include the `entity` parameter specifying the entity associated with this Index Definition. It is recommended to have one Index Definition per searchable entity.
 
 [source,java,indent=0]
 ----
 include::example$/search-ex1/src/main/java/com/company/demo/service/OrderIndexDefinition.java[tags=index-definition]
 ----
 
+=== `@JmixEntitySearchIndex`
```
Comment (knstvk) [inline content/modules/search/pages/index-definitions.adoc:19]: Add an anchor

File: content/modules/search/pages/index.adoc [hunk 7]
```diff
@@ -17,16 +19,28 @@ For manual installation, add the following dependencies to your `build.gradle`:
...
-[[connecting-to-elasticsearch-service]]
-== Connecting to Elasticsearch Service
+By default, add-on will be initially installed with OpenSearch support.
+
+If you upgrade project from the previous Jmix version with installed add-on - it will keep Elasticsearch installation.
+
+If you want to switch between Elasticsearch and OpenSearch - replace `io.jmix.search:jmix-search-opensearch-starter` with `io.jmix.search:jmix-search-elasticsearch-starter` or vice versa.
+
+[[connecting-to-search-engine-service]]
+== Connecting to search engine service
```
Comment (knstvk) [inline content/modules/search/pages/index.adoc]: Use Title Case

File: content/modules/search/pages/index.adoc [hunk 8]
```diff
@@ -17,16 +19,28 @@ For manual installation, add the following dependencies to your `build.gradle`:
...
+
+To connect to the Elasticsearch\OpenSearch service, specify the following properties in the `application.properties` file:
+
+* `jmix.search.server.url` - the complete URL of the service.
+* `jmix.search.server.login` - the login for basic authentication.
+* `jmix.search.server.password` - the user password used to connect to the service.
+
+== Versions compatibility
```
Comment (knstvk) [inline content/modules/search/pages/index.adoc]: Use title case, add an anchor

File: content/modules/search/pages/index.adoc [hunk 9]
```diff
@@ -17,16 +19,28 @@ For manual installation, add the following dependencies to your `build.gradle`:
...
-To connect to the Elasticsearch service, specify the following properties in the `application.properties` file:
+Java API clients for Elasticsearch\OpenSearch are forward-compatible with server version.
 
-* `jmix.search.elasticsearch.url` - the complete URL of the Elasticsearch service.
-* `jmix.search.elasticsearch.login` - the Elasticsearch login for basic authentication.
-* `jmix.search.elasticsearch.password` - the user password used to connect to the Elasticsearch service.
+Search add-on uses Elasticsearch Java API client v8.10.4 that supports ES 8.10+ (but also should support older versions till the oldest version is still supported by Elastic - 7.17.x)
 
-By default, the Search add-on is configured for Elasticsearch version 7.11 and higher. If you intend to utilize a version prior to 7.11, you can disable compatibility mode by setting the xref:search-properties.adoc#jmix.search.restHighLevelClientApiCompatibilityModeEnabled[jmix.search.restHighLevelClientApiCompatibilityModeEnabled] application property to `false`.
+If case of OpenSearch add-on uses Java API client v2.10.3.
```
Comment (knstvk) [inline content/modules/search/pages/index.adoc]: "With OpenSearch, the add-on uses ..."

File: content/modules/search/pages/search-in-ui.adoc [hunk 10]
```diff
@@ -96,11 +96,22 @@ To override the default strategy, add the following property to your `applicatio
 include::example$/search-ex1/src/main/resources/application.properties[tags=default-strategy]
 ----
 
-Additionally, you can create a custom search strategy. For that purpose, you need to create a new bean extending the `io.jmix.search.searching.SearchStrategy` interface and implement two methods:
+==== Custom Search Strategies
```
Comment (knstvk) [inline content/modules/search/pages/search-in-ui.adoc:100]: Add an anchor

**Review by knstvk — CHANGES_REQUESTED** [review]

File: content/modules/search/pages/config-options.adoc (hunk 1 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/config-options.adoc]: Done

File: content/modules/search/pages/config-options.adoc (hunk 2 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/config-options.adoc]: Done

File: content/modules/search/pages/config-options.adoc (hunk 3 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/config-options.adoc:94]: Done

File: content/modules/search/pages/config-options.adoc (hunk 4 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/config-options.adoc]: Done

File: content/modules/search/pages/config-options.adoc (hunk 5 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/config-options.adoc]: Done

File: content/modules/search/pages/index-definitions.adoc (hunk 6 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/index-definitions.adoc:19]: Done

File: content/modules/search/pages/index.adoc (hunk 7 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/index.adoc]: Done

File: content/modules/search/pages/index.adoc (hunk 8 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/index.adoc]: Done

File: content/modules/search/pages/index.adoc (hunk 9 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/index.adoc]: Done

File: content/modules/search/pages/search-in-ui.adoc (hunk 10 above)
Comment (Gavrilov-Ivan) [reply content/modules/search/pages/search-in-ui.adoc:100]: Done

**Pull request merged**